java-frontend/src/main/java/org/sonar/java/se/symbolicvalues/BinarySymbolicValue.java

```diff
@@ -47,8 +47,12 @@ public boolean references(SymbolicValue other) {
   @Override
   public void computedFrom(List<SymbolicValue> symbolicValues) {
     Preconditions.checkArgument(symbolicValues.size() == 2);
-    rightOp = symbolicValues.get(0);
-    leftOp = symbolicValues.get(1);
+    if (rightOp == null && leftOp == null) {
+      rightOp = symbolicValues.get(0);
+      leftOp = symbolicValues.get(1);
+    } else {
+      // no op - symbolic value has been reused and should not be recomputed
+    }
   }
 
   protected List<ProgramState> copyConstraint(SymbolicValue from, SymbolicValue to, ProgramState programState, BooleanConstraint booleanConstraint) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) The Cyclomatic Complexity of this method "copyConstraint" is 12 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)